lib/ansible/modules/cloud/ovh/ovh_domain_zone_record.py

```diff
@@ -0,0 +1,245 @@
+        except OvhApiError as apiError:
+            module.fail_json(
+                msg='Unable to call OVH api for getting the property of a record '
+                    'of the domain zone. Error returned by OVH api was : {0}'
+                    .format(apiError))
+        recordProperties.append(recordProperty)
+        if recordProperty['target'] in newTargets:
+            newTargets.remove(recordProperty['target'])
+
+    moduleChanged = False
+    for recordProperty in recordProperties:
+        if recordProperty['target'] in targets:
+            if state == "absent":
+                # Delete record
+                try:
+                    ovhApi.delete('/domain/zone/{0}/record/{1}'.format(name, recordProperty['id']))
+                except OvhApiError as apiError:
+                    module.fail_json(
+                        msg='Unable to call OVH api for deleting the record. '
+                            'Error returned by OVH api was : {0}'.format(apiError))
+                moduleChanged = True
+            else:
+                if recordProperty['ttl'] != ttl:
+                    # Update ttl
+                    try:
+                        ovhApi.put(
+                            '/domain/zone/{0}/record/{1}'
+                            .format(name, recordProperty['id']), ttl=ttl)
+                    except OvhApiError as apiError:
+                        module.fail_json(
+                            msg='Unable to call OVH api for updating the ttl of the '
+                                'record. Error returned by OVH api was : {0}'
+                                .format(apiError))
+                    moduleChanged = True
+        else:
+            if state == "present":
+                # Delete record (target change)
+                try:
+                    ovhApi.delete('/domain/zone/{0}/record/{1}'.format(name, recordProperty['id']))
+                except OvhApiError as apiError:
+                    module.fail_json(
+                        msg='Unable to call OVH api for deleting the record. '
+                            'Error returned by OVH api was : {0}'.format(apiError))
+                moduleChanged = True
+
+    for newTarget in newTargets:
+        if state == "present":
+            # Add record
+            try:
+                ovhApi.post(
+                    '/domain/zone/{0}/record'.format(name), fieldType=record_type,
+                    subDomain=subdomain, target=newTarget, ttl=ttl)
+            except OvhApiError as apiError:
+                module.fail_json(
+                    msg='Unable to call OVH api for adding the record, '
+                        'Error returned by OVH api was : {0}'.format(apiError))
+            moduleChanged = True
+
+    if moduleChanged:
+        # Apply zone modification
+        try:
+            ovhApi.post('/domain/zone/{0}/refresh'.format(name))
+            ovhApi.wait_for_no_task('/domain/zone/{0}'.format(name))
```

Comment:
There is no task system on the dns zone record resource (the /domain/{}/task endpoint is for another resource). We just have to hit the refresh endpoint (which return nothing, no task id) to enable the dns record.
We can see this by observing the api http calls in the ovh manager while adding dns records to a zone.